src/Plugin/jsonapi/FieldEnhancer/CardLinkEnhancer.php

```diff
@@ -84,8 +84,13 @@ public function getAlias($nid) {
    *   The array of fields value.
    */
   public function getCardFields($nid) {
-    $card_fields = [];
     $node = \Drupal::entityTypeManager()->getStorage('node')->load($nid);
+
```

Comment:
Hi @MdNadimHossain , I think you should ensure the node exists from this line https://github.com/dpc-sdp/tide_landing_page/blob/9b4219dfbab777d4742540227679407cecd2b9e1/src/Plugin/jsonapi/FieldEnhancer/CardLinkEnhancer.php#L29 .

Reply:
if this is not the case, please consider this regex.
`/^(internal\:\/node\/)|^(entity:node\/)/` , this will help you to check if the uri matches the desired shape. 
then you can extract node id by using 
`substr($uri, strrpos($uri, '/') + 1)`
then check is the result is a number.

Reply:
I have added new cards with internal links and looks like the uri returns like this `"uri": "entity:node/20",`
So I am just leaving it like this.
![Screen Shot 2021-02-24 at 12 34 52 pm](https://user-images.githubusercontent.com/20810541/108932098-e8f0c100-769c-11eb-96ad-2c1ffe179ede.png)
![Screen Shot 2021-02-24 at 12 35 16 pm](https://user-images.githubusercontent.com/20810541/108932132-f73edd00-769c-11eb-93f0-ab1197eb98c6.png)